src/amqpcat.cr

```diff
@@ -131,6 +141,37 @@ class AMQPCat
     end
   end
 
+  private def write_message_to_file(msg, dir_path : String)
+    # use message ID if available, or correlation ID, otherwise use timestamp
+    msg_id = msg.properties.message_id || msg.properties.correlation_id || "#{Time.utc.to_unix_ns}"
+
+    # get appropriate extension based on message properties
+    extension = guess_extension(msg.properties.content_type, msg.properties.content_encoding)
+
+    file_path = File.join(dir_path, "#{msg_id}#{extension}")
+
+    File.open(file_path, "w") do |file|
+      decode_payload(msg, file)
+    end
+
+    STDOUT.puts "#{file_path}"
+  end
+
+  private def guess_extension(content_type : String?, content_encoding : String? = nil) : String
+    # content_type is preferred over content_encoding
+    if content_type
+      extensions = MIME.extensions(content_type)
+      return "#{extensions.first}" unless extensions.empty?
+    end
+
+    if content_encoding == "gzip" || content_encoding == "deflate"
```

Comment:
Would it be of interest to unzip of inflate the content before writing it to file?

Reply:
Maybe as a separate PR, just a note to make you aware that it exists in the stdlib